diff --git a/src/GameApplication.php b/src/GameApplication.php
index 212f4e8..329859b 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -3,6 +3,8 @@
 namespace App;
 
 use App\ActionCommand\AttackCommand;
+use App\ActionCommand\HealCommand;
+use App\ActionCommand\SurrenderCommand;
 use App\Builder\CharacterBuilder;
 use App\Character\Character;
 use App\Observer\GameObserverInterface;
@@ -34,8 +36,19 @@ class GameApplication
             ), '']);
 
             // Player's turn
-            $attackCommand = new AttackCommand($player, $ai, $fightResultSet);
-            $attackCommand->execute();
+            $actionChoice = GameApplication::$printer->choice('Your turn', [
+                'Attack',
+                'Heal',
+                'Surrender',
+            ]);
+
+            $playerAction = match ($actionChoice) {
+                'Attack' => new AttackCommand($player, $ai, $fightResultSet),
+                'Heal' => new HealCommand($player),
+                'Surrender' => new SurrenderCommand($player),
+            };
+
+            $playerAction->execute();
 
             if ($this->didPlayerDie($ai)) {
                 $this->endBattle($fightResultSet, $player, $ai);
