diff --git a/src/GameApplication.php b/src/GameApplication.php
index f5a09bb..2313bfa 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -2,6 +2,7 @@
 
 namespace App;
 
+use App\ActionCommand\AttackCommand;
 use App\Builder\CharacterBuilder;
 use App\Character\Character;
 use App\Observer\GameObserverInterface;
@@ -26,18 +27,8 @@ class GameApplication
             ), '']);
 
             // Player's turn
-            $damage = $player->attack();
-            if ($damage === 0) {
-                self::$printer->printFor($player)->exhaustedMessage();
-                $fightResultSet->of($player)->addExhaustedTurn();
-            }
-
-            $damageDealt = $ai->receiveAttack($damage);
-            $fightResultSet->of($player)->addDamageDealt($damageDealt);
-
-            self::$printer->printFor($player)->attackMessage($damageDealt);
-            self::$printer->writeln('');
-            usleep(300000);
+            $attackCommand = new AttackCommand($player, $ai, $fightResultSet);
+            $attackCommand->execute();
 
             if ($this->didPlayerDie($ai)) {
                 $this->endBattle($fightResultSet, $player, $ai);
@@ -45,18 +36,8 @@ class GameApplication
             }
 
             // AI's turn
-            $aiDamage = $ai->attack();
-
-            if ($damage === 0) {
-                self::$printer->printFor($ai)->exhaustedMessage();
-                $fightResultSet->of($ai)->addExhaustedTurn();
-            }
-
-            $damageReceived = $player->receiveAttack($aiDamage);
-            $fightResultSet->of($ai)->addDamageReceived($damageReceived);
-
-            self::$printer->printFor($ai)->attackMessage($damageReceived);
-            self::$printer->writeln('');
+            $aiAttackCommand = new AttackCommand($ai, $player, $fightResultSet);
+            $aiAttackCommand->execute();
 
             if ($this->didPlayerDie($player)) {
                 $this->endBattle($fightResultSet, $ai, $player);
