diff --git a/src/ActionCommand/AttackCommand.php b/src/ActionCommand/AttackCommand.php
index 7d1b414..8a1cff3 100644
--- a/src/ActionCommand/AttackCommand.php
+++ b/src/ActionCommand/AttackCommand.php
@@ -17,20 +17,19 @@ class AttackCommand implements ActionCommandInterface
 
     public function execute(): void
     {
-        $damage = $player->attack();
+        $damage = $this->player->attack();
         if ($damage === 0) {
-            GameApplication::$io->writeln('You\'re exhausted for a turn');
-            $fightResult->addExhaustedTurn();
+            GameApplication::$printer->printFor($this->player)->exhaustedMessage();
+            $this->fightResult->addExhaustedTurn($this->player);
+
+            return;
         }
 
-        $damageDealt = $ai->receiveAttack($damage);
-        $fightResult->addDamageDealt($damageDealt);
+        $damageDealt = $this->opponent->receiveAttack($damage);
+        $this->fightResult->addDamageDealt($this->player, $damageDealt);
 
-        GameApplication::$io->writeln(sprintf(
-            'Your attack does %d damage',
-            $damageDealt
-        ));
-        GameApplication::$io->writeln('');
+        GameApplication::$printer->printFor($this->player)->attackMessage($damageDealt);
+        GameApplication::$printer->writeln('');
         usleep(300000);
     }
 }
