diff --git a/src/Character/Character.php b/src/Character/Character.php
index 36aa595..7d04e8a 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -20,7 +20,7 @@ class Character
         private AttackType $attackType,
         private ArmorType $armorType
     ) {
-        $this->currentHealth = $this->maxHealth;
+        $this->currentHealth = $maxHealth;
     }
 
     public function attack(): int
