diff --git a/src/GameApplication.php b/src/GameApplication.php
index d00ed8f..d04e500 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -2,6 +2,7 @@
 
 namespace App;
 
+use App\Builder\CharacterBuilder;
 use App\ArmorType\IceBlockType;
 use App\ArmorType\LeatherArmorType;
 use App\ArmorType\ShieldType;
@@ -74,4 +75,9 @@ class GameApplication
     {
         return $player->getCurrentHealth() <= 0;
     }
+
+    private function createCharacterBuilder(): CharacterBuilder
+    {
+        return new CharacterBuilder();
+    }
 }
