diff --git a/src/GameApplication.php b/src/GameApplication.php
index abe3b02..b5ed0a5 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -2,6 +2,7 @@
 
 namespace App;
 
+use App\Builder\CharacterBuilder;
 use App\ArmorType\IceBlockType;
 use App\ArmorType\LeatherArmorType;
 use App\ArmorType\ShieldType;
@@ -74,4 +75,9 @@ class GameApplication
             'mage_archer'
         ];
     }
+
+    private function createCharacterBuilder(): CharacterBuilder
+    {
+        return new CharacterBuilder();
+    }
 }
