diff --git a/src/Character/Archer.php b/src/Character/Archer.php
deleted file mode 100644
index b8a4d65..0000000
--- a/src/Character/Archer.php
+++ /dev/null
@@ -1,14 +0,0 @@
-<?php
-
-namespace App\Character;
-
-use App\ArmorType\LeatherArmorType;
-use App\AttackType\BowType;
-
-class Archer extends Character
-{
-    public function __construct()
-    {
-        parent::__construct(80, 10, new BowType(), new LeatherArmorType());
-    }
-}
diff --git a/src/Character/Fighter.php b/src/Character/Fighter.php
deleted file mode 100644
index 75081b9..0000000
--- a/src/Character/Fighter.php
+++ /dev/null
@@ -1,14 +0,0 @@
-<?php
-
-namespace App\Character;
-
-use App\ArmorType\ShieldType;
-use App\AttackType\TwoHandedSwordType;
-
-class Fighter extends Character
-{
-    public function __construct()
-    {
-        parent::__construct(90, 12, new TwoHandedSwordType(), new ShieldType());
-    }
-}
diff --git a/src/Character/Mage.php b/src/Character/Mage.php
deleted file mode 100644
index cb92ad9..0000000
--- a/src/Character/Mage.php
+++ /dev/null
@@ -1,14 +0,0 @@
-<?php
-
-namespace App\Character;
-
-use App\ArmorType\IceBlockType;
-use App\AttackType\FireBoltType;
-
-class Mage extends Character
-{
-    public function __construct()
-    {
-        parent::__construct(70, 8, new FireBoltType(), new IceBlockType());
-    }
-}
diff --git a/src/Character/MageFighter.php b/src/Character/MageFighter.php
deleted file mode 100644
index 33dd300..0000000
--- a/src/Character/MageFighter.php
+++ /dev/null
@@ -1,14 +0,0 @@
-<?php
-
-namespace App\Character;
-
-use App\ArmorType\ShieldType;
-use App\AttackType\FireBoltType;
-
-class MageFighter extends Character
-{
-    public function __construct()
-    {
-        parent::__construct(100, 12, new FireBoltType(), new ShieldType());
-    }
-}
