diff --git a/src/Character/Character.php b/src/Character/Character.php
index 51c8a1d..d7b2b52 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -5,7 +5,7 @@ namespace App\Character;
 use App\ArmorType\ArmorType;
 use App\AttackType\AttackType;
 
-abstract class Character
+class Character
 {
     private int $currentHealth;
     private string $nickname = '';
@@ -16,7 +16,7 @@ abstract class Character
         private AttackType $attackType,
         private ArmorType $armorType
     ) {
-        $this->currentHealth = $this->maxHealth;
+        $this->currentHealth = $maxHealth;
     }
 
     /**
