diff --git a/src/GameApplication.php b/src/GameApplication.php
index d04e500..24415ed 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -3,13 +3,6 @@
 namespace App;
 
 use App\Builder\CharacterBuilder;
-use App\ArmorType\IceBlockType;
-use App\ArmorType\LeatherArmorType;
-use App\ArmorType\ShieldType;
-use App\AttackType\BowType;
-use App\AttackType\FireBoltType;
-use App\AttackType\MultiAttackType;
-use App\AttackType\TwoHandedSwordType;
 use App\Character\Character;
 
 class GameApplication
@@ -46,10 +39,35 @@ class GameApplication
     public function createCharacter(string $character): Character
     {
         return match (strtolower($character)) {
-            'fighter' => new Character(90, 12, new TwoHandedSwordType(), new ShieldType()),
-            'archer' => new Character(80, 10, new BowType(), new LeatherArmorType()),
-            'mage' => new Character(70, 8, new FireBoltType(), new IceBlockType()),
-            'mage_archer' => new Character(75, 9, new MultiAttackType([new BowType(), new FireBoltType()]), new ShieldType()),
+            'fighter' => $this->createCharacterBuilder()
+                ->setMaxHealth(90)
+                ->setBaseDamage(12)
+                ->setAttackType('sword')
+                ->setArmorType('shield')
+                ->buildCharacter(),
+
+            'archer' => $this->createCharacterBuilder()
+                ->setMaxHealth(80)
+                ->setBaseDamage(10)
+                ->setAttackType('bow')
+                ->setArmorType('leather_armor')
+                ->buildCharacter(),
+
+            'mage' => $this->createCharacterBuilder()
+                ->setMaxHealth(70)
+                ->setBaseDamage(8)
+                ->setAttackType('fire_bolt')
+                ->setArmorType('ice_block')
+                ->buildCharacter(),
+
+            'mage_archer' => $this->createCharacterBuilder()
+                ->setMaxHealth(90)
+                ->setBaseDamage(12)
+                ->setAttackType('fire_bolt') // TODO re-add bow!
+                ->setArmorType('shield')
+                ->buildCharacter(),
+
+            default => throw new \RuntimeException('Undefined Character')
         };
     }
 
