diff --git a/src/Builder/CharacterBuilder.php b/src/Builder/CharacterBuilder.php
index 5755976..e595f85 100644
--- a/src/Builder/CharacterBuilder.php
+++ b/src/Builder/CharacterBuilder.php
@@ -2,6 +2,32 @@
 
 namespace App\Builder;
 
+use App\Character\Character;
+
 class CharacterBuilder
 {
+    public function setMaxHealth(int $maxHealth): self
+    {
+        // todo: implement me
+    }
+
+    public function setBaseDamage(int $baseDamage): self
+    {
+        // todo: implement me
+    }
+
+    public function setAttackType(string $attackType): self
+    {
+        // todo: implement me
+    }
+
+    public function setArmorType(string $armorType): self
+    {
+        // todo: implement me
+    }
+
+    public function buildCharacter(): Character
+    {
+        // todo: implement me
+    }
 }
