diff --git a/src/Decorator/DebugEventDispatcherDecorator.php b/src/Decorator/DebugEventDispatcherDecorator.php
new file mode 100644
index 0000000..011bd77
--- /dev/null
+++ b/src/Decorator/DebugEventDispatcherDecorator.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Decorator;
+
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class DebugEventDispatcherDecorator implements EventDispatcherInterface
+{
+    public function __construct(
+        private readonly EventDispatcherInterface $eventDispatcher
+    ) {
+    }
+}
