diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
index 1c8bd3a..a1e64e7 100644
--- a/src/Service/OutputtingXpCalculator.php
+++ b/src/Service/OutputtingXpCalculator.php
@@ -2,10 +2,16 @@
 
 namespace App\Service;
 
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+
 class OutputtingXpCalculator implements XpCalculatorInterface
 {
+    private OutputInterface $output;
+
     public function __construct(
         private readonly XpCalculatorInterface $innerCalculator
     ) {
+        $this->output = new ConsoleOutput();
     }
 }
