diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
index fedb85c..58da840 100644
--- a/src/Service/OutputtingXpCalculator.php
+++ b/src/Service/OutputtingXpCalculator.php
@@ -4,7 +4,9 @@ namespace App\Service;
 
 use App\Character\Character;
 use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
 
+#[AsDecorator(XpCalculatorInterface::class)]
 class OutputtingXpCalculator implements XpCalculatorInterface
 {
     public function __construct(
