diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
index 1f40b29..fedb85c 100644
--- a/src/Service/OutputtingXpCalculator.php
+++ b/src/Service/OutputtingXpCalculator.php
@@ -3,6 +3,7 @@
 namespace App\Service;
 
 use App\Character\Character;
+use Symfony\Component\Console\Output\ConsoleOutput;
 
 class OutputtingXpCalculator implements XpCalculatorInterface
 {
@@ -14,6 +15,17 @@ class OutputtingXpCalculator implements XpCalculatorInterface
 
     public function addXp(Character $winner, int $enemyLevel): void
     {
+        $beforeLevel = $winner->getLevel();
+
         $this->innerCalculator->addXp($winner, $enemyLevel);
+
+        $afterLevel = $winner->getLevel();
+        if ($afterLevel > $beforeLevel) {
+            $output = new ConsoleOutput();
+            $output->writeln('--------------------------------');
+            $output->writeln('<bg=green;fg=white>Congratulations! You\'ve leveled up!</>');
+            $output->writeln(sprintf('You are now level "%d"', $winner->getLevel()));
+            $output->writeln('--------------------------------');
+        }
     }
 }
