diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
new file mode 100644
index 0000000..1c8bd3a
--- /dev/null
+++ b/src/Service/OutputtingXpCalculator.php
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Service;
+
+class OutputtingXpCalculator implements XpCalculatorInterface
+{
+    public function __construct(
+        private readonly XpCalculatorInterface $innerCalculator
+    ) {
+    }
+}
