diff --git a/src/Decorator/DebugEventDispatcherDecorator.php b/src/Decorator/DebugEventDispatcherDecorator.php
index 5f70ca5..f2538ae 100644
--- a/src/Decorator/DebugEventDispatcherDecorator.php
+++ b/src/Decorator/DebugEventDispatcherDecorator.php
@@ -2,9 +2,11 @@
 
 namespace App\Decorator;
 
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
+#[AsDecorator('event_dispatcher')]
 class DebugEventDispatcherDecorator implements EventDispatcherInterface
 {
     public function __construct(
