diff --git a/src/Decorator/OutputtingXpEarnedServiceDecorator.php b/src/Decorator/OutputtingXpEarnedServiceDecorator.php
new file mode 100644
index 0000000..4d3aa8c
--- /dev/null
+++ b/src/Decorator/OutputtingXpEarnedServiceDecorator.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Decorator;
+
+use App\Character\Character;
+
+class OutputtingXpEarnedServiceDecorator implements XpEarnedServiceInterface
+{
+    public function __construct(
+        private readonly XpEarnedServiceInterface $xpEarnedService
+    ) {
+    }
+}
