diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
index 1de1ac5..4065978 100644
--- a/src/Service/OutputtingXpCalculator.php
+++ b/src/Service/OutputtingXpCalculator.php
@@ -18,6 +18,16 @@ class OutputtingXpCalculator implements XpCalculatorInterface
 
     public function addXp(Character $winner, int $enemyLevel): void
     {
+        $beforeLevel = $winner->getLevel();
+
         $this->innerCalculator->addXp($winner, $enemyLevel);
+
+        $afterLevel = $winner->getLevel();
+        if ($afterLevel > $beforeLevel) {
+            $this->output->writeln('--------------------------------');
+            $this->output->writeln('<bg=green;fg=white>Congratulations! You\'ve leveled up!</>');
+            $this->output->writeln(sprintf('You are now level "%d"', $winner->getLevel()));
+            $this->output->writeln('--------------------------------');
+        }
     }
 }
