diff --git a/src/Decorator/XpEarnedServiceInterface.php b/src/Decorator/XpEarnedServiceInterface.php
new file mode 100644
index 0000000..364c757
--- /dev/null
+++ b/src/Decorator/XpEarnedServiceInterface.php
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Decorator;
+
+use App\Character\Character;
+
+interface XpEarnedServiceInterface
+{
+    public function addXp(Character $winner, int $enemyLevel): void;
+}
