diff --git a/src/Service/OutputtingXpCalculator.php b/src/Service/OutputtingXpCalculator.php
index 924ba6a..1f40b29 100644
--- a/src/Service/OutputtingXpCalculator.php
+++ b/src/Service/OutputtingXpCalculator.php
@@ -2,6 +2,8 @@
 
 namespace App\Service;
 
+use App\Character\Character;
+
 class OutputtingXpCalculator implements XpCalculatorInterface
 {
     public function __construct(
@@ -9,4 +11,9 @@ class OutputtingXpCalculator implements XpCalculatorInterface
     )
     {
     }
+
+    public function addXp(Character $winner, int $enemyLevel): void
+    {
+        $this->innerCalculator->addXp($winner, $enemyLevel);
+    }
 }
