diff --git a/src/Observer/XpEarnedObserver.php b/src/Observer/XpEarnedObserver.php
index 5dc9392..cafa88b 100644
--- a/src/Observer/XpEarnedObserver.php
+++ b/src/Observer/XpEarnedObserver.php
@@ -3,12 +3,12 @@
 namespace App\Observer;
 
 use App\FightResult;
-use App\Service\XpCalculator;
+use App\Service\XpCalculatorInterface;
 
 class XpEarnedObserver implements GameObserverInterface
 {
     public function __construct(
-        private readonly XpCalculator $xpCalculator
+        private readonly XpCalculatorInterface $xpCalculator
     ) {
     }
 
