diff --git a/src/GameApplication.php b/src/GameApplication.php
index a2a141a..edd08cf 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -9,6 +9,7 @@ use App\Builder\CharacterBuilder;
 use App\ChainHandler\CasinoHandler;
 use App\ChainHandler\XpBonusHandlerInterface;
 use App\Character\Character;
+use App\Factory\UltimateAttackTypeFactory;
 use App\Observer\GameObserverInterface;
 use App\Printer\MessagePrinter;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
@@ -30,6 +31,21 @@ class GameApplication
         $this->difficultyContext = new GameDifficultyContext();
     }
 
+    public function activateCheatCode(string $cheatCode): void
+    {
+        switch ($cheatCode) {
+            // Famous Konami Code
+            case 'up-up-down-down-left-right-left-right-b-a-start':
+                $this->characterBuilder->setAttackTypeFactory(new UltimateAttackTypeFactory());
+
+                GameApplication::$printer->info('Cheat code activated!!');
+                break;
+            default:
+                GameApplication::$printer->info('Invalid cheat code');
+                break;
+        }
+    }
+
     public function play(Character $player, Character $ai, FightResultSet $fightResultSet): void
     {
         while (true) {
