diff --git a/src/Kernel.php b/src/Kernel.php
index 779cd1f..e75623e 100644
--- a/src/Kernel.php
+++ b/src/Kernel.php
@@ -2,10 +2,18 @@
 
 namespace App;
 
+use App\Observer\GameObserverInterface;
 use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\HttpKernel\Kernel as BaseKernel;
 
 class Kernel extends BaseKernel
 {
     use MicroKernelTrait;
+
+    protected function build(ContainerBuilder $container)
+    {
+        $container->registerForAutoconfiguration(GameObserverInterface::class)
+            ->addTag('game.observer');
+    }
 }
