diff --git a/src/GameApplication.php b/src/GameApplication.php
index 54013bc..1870269 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -4,6 +4,7 @@ namespace App;
 
 use App\Builder\CharacterBuilder;
 use App\Character\Character;
+use App\Observer\GameObserverInterface;
 
 class GameApplication
 {
@@ -81,6 +82,16 @@ class GameApplication
         ];
     }
 
+    public function subscribe(GameObserverInterface $observer): void
+    {
+        // TODO: Implement subscribe() method.
+    }
+
+    public function unsubscribe(GameObserverInterface $observer): void
+    {
+        // TODO: Implement unsubscribe() method.
+    }
+
     private function finishFightResult(FightResult $fightResult, Character $winner, Character $loser): FightResult
     {
         $fightResult->setWinner($winner);
