diff --git a/src/GameApplication.php b/src/GameApplication.php
index 50d1d1d..0b140ec 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -111,6 +111,8 @@ class GameApplication
         $fightResult->setWinner($winner);
         $fightResult->setLoser($loser);
 
+        $this->notify($fightResult);
+
         return $fightResult;
     }
 
@@ -123,4 +125,11 @@ class GameApplication
     {
         return $this->characterBuilderFactory->createBuilder();
     }
+
+    private function notify(FightResult $fightResult): void
+    {
+        foreach ($this->observers as $observer) {
+            $observer->onFightFinished($fightResult);
+        }
+    }
 }
