diff --git a/src/Command/GameCommand.php b/src/Command/GameCommand.php
index e25f89b..c723880 100644
--- a/src/Command/GameCommand.php
+++ b/src/Command/GameCommand.php
@@ -98,6 +98,7 @@ class GameCommand extends Command
         $io->writeln('Total Rounds: ' . $fightResult->getRounds());
         $io->writeln('Damage dealt: ' . $fightResult->getDamageDealt());
         $io->writeln('Damage received: ' . $fightResult->getDamageReceived());
+        $io->writeln('Final Level: ' . $player->getLevel());
         $io->writeln('Exhausted Turns:' . $fightResult->getExhaustedTurns());
         $io->writeln('------------------------------');
     }
