diff --git a/src/Character/Character.php b/src/Character/Character.php
index bc123bd..4f56198 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -54,6 +54,18 @@ class Character
         $this->nickname = $nickname;
     }
 
+    public function levelUp(): void
+    {
+        // +%15 bonus to stats
+        $bonus = 1.15;
+
+        $this->level++;
+        $this->maxHealth *= $bonus;
+        $this->baseDamage *= $bonus;
+
+        // todo: level up attack and armor type
+    }
+
     public function getLevel(): int
     {
         return $this->level;
