diff --git a/src/Observer/XpEarnedObserver.php b/src/Observer/XpEarnedObserver.php
new file mode 100644
index 0000000..b4ff7f3
--- /dev/null
+++ b/src/Observer/XpEarnedObserver.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Observer;
+
+use App\FightResult;
+
+class XpEarnedObserver implements GameObserverInterface
+{
+    public function onFightFinished(FightResult $fightResult): void
+    {
+        // TODO: Implement onFightFinished() method.
+    }
+}
