diff --git a/config/services.yaml b/config/services.yaml
index 2d6a76f..e268993 100644
--- a/config/services.yaml
+++ b/config/services.yaml
@@ -22,3 +22,7 @@ services:
 
     # add more service definitions when explicit configuration is needed
     # please note that last definitions always *replace* previous ones
+
+    App\GameApplication:
+        calls:
+            - subscribe: ['@App\Observer\XpEarnedObserver']
diff --git a/src/Command/GameCommand.php b/src/Command/GameCommand.php
index 1a9214f..b3ea7e0 100644
--- a/src/Command/GameCommand.php
+++ b/src/Command/GameCommand.php
@@ -5,8 +5,6 @@ namespace App\Command;
 use App\Character\Character;
 use App\FightResult;
 use App\GameApplication;
-use App\Observer\XpEarnedObserver;
-use App\Service\XpCalculator;
 use Symfony\Component\Console\Attribute\AsCommand;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
@@ -25,11 +23,6 @@ class GameCommand extends Command
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $xpObserver = new XpEarnedObserver(
-            new XpCalculator()
-        );
-        $this->game->subscribe($xpObserver);
-
         $io = new SymfonyStyle($input, $output);
 
         $io->text('Welcome to the game where warriors fight against each other for honor and glory... and 🍕!');
