diff --git a/src/GameApplication.php b/src/GameApplication.php
index 1870269..30bddcc 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -8,6 +8,9 @@ use App\Observer\GameObserverInterface;
 
 class GameApplication
 {
+    /** @var GameObserverInterface[] */
+    private array $observers = [];
+
     public function play(Character $player, Character $ai): FightResult
     {
         $player->rest();
@@ -84,12 +87,18 @@ class GameApplication
 
     public function subscribe(GameObserverInterface $observer): void
     {
-        // TODO: Implement subscribe() method.
+        if (!in_array($observer, $this->observers, true)) {
+            $this->observers[] = $observer;
+        }
     }
 
     public function unsubscribe(GameObserverInterface $observer): void
     {
-        // TODO: Implement unsubscribe() method.
+        $key = array_search($observer, $this->observers, true);
+
+        if ($key !== false) {
+            unset($this->observers[$key]);
+        }
     }
 
     private function finishFightResult(FightResult $fightResult, Character $winner, Character $loser): FightResult
