diff --git a/src/Observer/GameSubjectInterface.php b/src/Observer/GameSubjectInterface.php
new file mode 100644
index 0000000..cb9f1c4
--- /dev/null
+++ b/src/Observer/GameSubjectInterface.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Observer;
+
+use App\FightResult;
+
+interface GameSubjectInterface
+{
+    public function subscribe(GameObserverInterface $observer): void;
+
+    public function unsubscribe(GameObserverInterface $observer): void;
+
+    public function notify(FightResult $fightResult): void;
+}
