diff --git a/src/Observer/XpEarnedObserver.php b/src/Observer/XpEarnedObserver.php
index b4ff7f3..ba89be6 100644
--- a/src/Observer/XpEarnedObserver.php
+++ b/src/Observer/XpEarnedObserver.php
@@ -3,11 +3,20 @@
 namespace App\Observer;
 
 use App\FightResult;
+use App\Service\XpEarnedService;
 
 class XpEarnedObserver implements GameObserverInterface
 {
+    public function __construct(
+        private readonly XpEarnedService $xpEarnedService
+    ) {
+    }
+
     public function onFightFinished(FightResult $fightResult): void
     {
-        // TODO: Implement onFightFinished() method.
+        $winner = $fightResult->getWinner();
+        $loser = $fightResult->getLoser();
+
+        $this->xpEarnedService->addXp($winner, $loser->getLevel());
     }
 }
