diff --git a/src/Command/GameCommand.php b/src/Command/GameCommand.php
index 4badc6a..a87585b 100644
--- a/src/Command/GameCommand.php
+++ b/src/Command/GameCommand.php
@@ -3,7 +3,6 @@
 namespace App\Command;
 
 use App\Character\Character;
-use App\Event\OutputFightStartingSubscriber;
 use App\FightResult;
 use App\GameApplication;
 use Symfony\Component\Console\Attribute\AsCommand;
@@ -11,14 +10,12 @@ use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
-use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 
 #[AsCommand('app:game:play')]
 class GameCommand extends Command
 {
     public function __construct(
         private readonly GameApplication $game,
-        private readonly EventDispatcherInterface $eventDispatcher,
     )
     {
         parent::__construct();
@@ -27,7 +24,6 @@ class GameCommand extends Command
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new SymfonyStyle($input, $output);
-        $this->eventDispatcher->addSubscriber(new OutputFightStartingSubscriber());
 
         $io->text('Welcome to the game where warriors fight against each other for honor and glory... and 🍕!');
 
