diff --git a/src/Command/GameCommand.php b/src/Command/GameCommand.php
index 2c8920f..9564dda 100644
--- a/src/Command/GameCommand.php
+++ b/src/Command/GameCommand.php
@@ -27,8 +27,8 @@ class GameCommand extends Command
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new SymfonyStyle($input, $output);
-        $this->eventDispatcher->addListener(FightStartingEvent::class, function() use ($io) {
-            $io->note('Fight is starting...');
+        $this->eventDispatcher->addListener(FightStartingEvent::class, function(FightStartingEvent $event) use ($io) {
+            $io->note(sprintf('Fight is starting against %s', $event->ai->getNickname()));
         });
 
         $io->text('Welcome to the game where warriors fight against each other for honor and glory... and 🍕!');
diff --git a/src/Event/FightStartingEvent.php b/src/Event/FightStartingEvent.php
index 38c56a1..64d8c1a 100644
--- a/src/Event/FightStartingEvent.php
+++ b/src/Event/FightStartingEvent.php
@@ -2,6 +2,11 @@
 
 namespace App\Event;
 
+use App\Character\Character;
+
 class FightStartingEvent
 {
+    public function __construct(public Character $player, public Character $ai)
+    {
+    }
 }
diff --git a/src/GameApplication.php b/src/GameApplication.php
index 76d47df..e45c785 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -24,7 +24,7 @@ class GameApplication
 
     public function play(Character $player, Character $ai): FightResult
     {
-        $this->eventDispatcher->dispatch(new FightStartingEvent());
+        $this->eventDispatcher->dispatch(new FightStartingEvent($player, $ai));
         $player->rest();
 
         $fightResult = new FightResult();
