diff --git a/src/GameApplication.php b/src/GameApplication.php
index 0b140ec..76d47df 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -5,19 +5,26 @@ namespace App;
 use App\Builder\CharacterBuilder;
 use App\Builder\CharacterBuilderFactory;
 use App\Character\Character;
+use App\Event\FightStartingEvent;
 use App\Observer\GameObserverInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 
 class GameApplication
 {
     /** @var GameObserverInterface[] */
     private array $observers = [];
 
-    public function __construct(private CharacterBuilderFactory $characterBuilderFactory)
+    public function __construct(
+        private CharacterBuilderFactory $characterBuilderFactory,
+        private EventDispatcherInterface $eventDispatcher,
+    )
     {
     }
 
     public function play(Character $player, Character $ai): FightResult
     {
+        $this->eventDispatcher->dispatch(new FightStartingEvent());
         $player->rest();
 
         $fightResult = new FightResult();
