diff --git a/src/Character/MageFighter.php b/src/Character/MageFighter.php
new file mode 100644
index 0000000..508e3db
--- /dev/null
+++ b/src/Character/MageFighter.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Character;
+
+use App\AttackType\CastSpellType;
+
+class MageFighter extends Character
+{
+    public function __construct()
+    {
+        $this->maxHealth = 100;
+        $this->currentHealth = $this->maxHealth;
+        $this->baseDamage = 12;
+        $this->armor = 0.25;
+
+        $this->attackType = new CastSpellType();
+    }
+}
