diff --git a/src/GameApplication.php b/src/GameApplication.php
index 9cf057b..507e216 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -4,6 +4,7 @@ namespace App;
 
 use App\AttackType\BowType;
 use App\AttackType\FireBoltType;
+use App\AttackType\MultiAttackType;
 use App\AttackType\TwoHandedSwordType;
 use App\Character\Character;
 
@@ -44,7 +45,7 @@ class GameApplication
             'fighter' => new Character(90, 12, 0.25, new TwoHandedSwordType()),
             'archer' => new Character(80, 10, 0.15, new BowType()),
             'mage' => new Character(70, 8, 0.10, new FireBoltType()),
-            default => throw new \RuntimeException('Undefined Character'),
+            'mage_archer' => new Character(75, 9, .15, new MultiAttackType([new BowType(), new FireBoltType()])),
         };
     }
 
@@ -54,6 +55,7 @@ class GameApplication
             'fighter',
             'mage',
             'archer',
+            'mage_archer'
         ];
     }
 
