diff --git a/src/Character/Mage.php b/src/Character/Mage.php
index ce82c46..523fb50 100644
--- a/src/Character/Mage.php
+++ b/src/Character/Mage.php
@@ -2,10 +2,25 @@
 
 namespace App\Character;
 
+use App\Dice;
+
 class Mage extends Character
 {
     public function __construct()
     {
         parent::__construct(70, 8, 0.10);
     }
+    
+    public function attack(): int
+    {
+        return $this->castSpell();
+    }
+
+    /**
+     * Damage: 3d10
+     */
+    private function castSpell(): int
+    {
+        return Dice::roll(10) + Dice::roll(10) + Dice::roll(10);
+    }
 }
