diff --git a/src/Character/IceMage.php b/src/Character/IceMage.php
new file mode 100644
index 0000000..9fb7c1f
--- /dev/null
+++ b/src/Character/IceMage.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Character;
+
+use App\Dice;
+
+/**
+ * Duplicate FireMage constructor or inherit from it (increasing inheritance level)
+ */
+class IceMage extends Mage
+{
+    public function attack(Character $target): int
+    {
+        return Dice::roll(4, 18);
+    }
+}
diff --git a/src/Character/MageFighter.php b/src/Character/MageFighter.php
new file mode 100644
index 0000000..5d89d8f
--- /dev/null
+++ b/src/Character/MageFighter.php
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Character;
+
+use App\Dice;
+
+class MageFighter extends Character
+{
+    public function __construct()
+    {
+        $this->health = 75;
+        $this->currentHealth = $this->health;
+        $this->baseDamage = 8;
+        $this->armor = 0.15;
+    }
+
+    /**
+     * MageFighter does not have iceBlock it uses regular armor
+     */
+    public function receiveAttack(int $enemyDamage): int
+    {
+        // duplicating code in root's base class
+        // it's becoming hard to maintain this design
+        $receivedDamage = $enemyDamage * (1 - $this->armor);
+        $this->currentHealth -= (int)$receivedDamage;
+
+        return $receivedDamage;
+    }
+
+    /**
+     * 30% chance of casting a spell on attack
+     */
+    public function attack(Character $target): int
+    {
+        // cannot reuse Fighter or Mage "attack()" method's code
+    }
+}
