diff --git a/src/AttackType/BowType.php b/src/AttackType/BowType.php
new file mode 100644
index 0000000..45d2ba3
--- /dev/null
+++ b/src/AttackType/BowType.php
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class BowType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        $criticalChance = Dice::roll(100);
+
+        return $criticalChance > 70 ? $baseDamage * 3 : $baseDamage;
+    }
+}
diff --git a/src/AttackType/FireBoltType.php b/src/AttackType/FireBoltType.php
new file mode 100644
index 0000000..e98ac6a
--- /dev/null
+++ b/src/AttackType/FireBoltType.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class FireBoltType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        return Dice::roll(10) + Dice::roll(10) + Dice::roll(10);
+    }
+}
diff --git a/src/AttackType/TwoHandedSwordType.php b/src/AttackType/TwoHandedSwordType.php
new file mode 100644
index 0000000..cb91732
--- /dev/null
+++ b/src/AttackType/TwoHandedSwordType.php
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class TwoHandedSwordType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        $twoHandledSwordDamage = Dice::roll(12) + Dice::roll(12);
+
+        return $baseDamage + $twoHandledSwordDamage;
+    }
+}
