diff --git a/src/Character/Character.php b/src/Character/Character.php
index 25ac7fd..1ed06d0 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -9,6 +9,8 @@ abstract class Character
     private int $currentHealth;
     private string $nickname = '';
 
+    abstract public function castSpell(): int;
+
     public function __construct(
         private int $maxHealth,
         private int $baseDamage,
