diff --git a/src/ArmorType/IceBlockType.php b/src/ArmorType/IceBlockType.php
new file mode 100644
index 0000000..9260f4e
--- /dev/null
+++ b/src/ArmorType/IceBlockType.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\ArmorType;
+
+use App\Dice;
+
+class IceBlockType implements ArmorType
+{
+    /**
+     * Absorbs 2d8
+     */
+    public function getArmorReduction(int $damage): int
+    {
+        return Dice::roll(8) + Dice::roll(8);
+    }
+}
diff --git a/src/ArmorType/LeatherArmorType.php b/src/ArmorType/LeatherArmorType.php
new file mode 100644
index 0000000..640c100
--- /dev/null
+++ b/src/ArmorType/LeatherArmorType.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\ArmorType;
+
+class LeatherArmorType implements ArmorType
+{
+    /**
+     * Absorbs 25% of the damage
+     */
+    public function getArmorReduction(int $damage): int
+    {
+        return floor($damage * 0.25);
+    }
+}
diff --git a/src/ArmorType/ShieldType.php b/src/ArmorType/ShieldType.php
new file mode 100644
index 0000000..d75a783
--- /dev/null
+++ b/src/ArmorType/ShieldType.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\ArmorType;
+
+use App\Dice;
+
+class ShieldType implements ArmorType
+{
+    /**
+     * Has 20% to fully block the attack
+     */
+    public function getArmorReduction(int $damage): int
+    {
+        $chanceToBlock = Dice::roll(100);
+
+        return $chanceToBlock > 80 ? $damage : 0;
+    }
+}
