diff --git a/src/Character/Archer.php b/src/Character/Archer.php
index ceb8173..41c4997 100644
--- a/src/Character/Archer.php
+++ b/src/Character/Archer.php
@@ -2,6 +2,7 @@
 
 namespace App\Character;
 
+use App\ArmorType\LeatherArmorType;
 use App\AttackType\ShootType;
 use App\Dice;
 
@@ -14,5 +15,6 @@ class Archer extends Character
         $this->baseDamage = 10;
         $this->armor = 0.15;
         $this->attackType = new ShootType();
+        $this->armorType = new LeatherArmorType();
     }
 }
diff --git a/src/Character/Fighter.php b/src/Character/Fighter.php
index 808c8f2..684f0c0 100644
--- a/src/Character/Fighter.php
+++ b/src/Character/Fighter.php
@@ -2,6 +2,7 @@
 
 namespace App\Character;
 
+use App\ArmorType\ShieldType;
 use App\AttackType\TwoHandedSwordType;
 
 class Fighter extends Character
@@ -13,5 +14,6 @@ class Fighter extends Character
         $this->baseDamage = 12;
         $this->armor = 0.25;
         $this->attackType = new TwoHandedSwordType();
+        $this->armorType = new ShieldType();
     }
 }
diff --git a/src/Character/Mage.php b/src/Character/Mage.php
index 20e72e8..eafd7d0 100644
--- a/src/Character/Mage.php
+++ b/src/Character/Mage.php
@@ -2,6 +2,7 @@
 
 namespace App\Character;
 
+use App\ArmorType\IceBlockType;
 use App\AttackType\CastSpellType;
 use App\Dice;
 
@@ -14,5 +15,6 @@ class Mage extends Character
         $this->baseDamage = 8;
         $this->armor = 0.10;
         $this->attackType = new CastSpellType();
+        $this->armorType = new IceBlockType();
     }
 }
diff --git a/src/Character/MageFighter.php b/src/Character/MageFighter.php
index 508e3db..8387490 100644
--- a/src/Character/MageFighter.php
+++ b/src/Character/MageFighter.php
@@ -2,6 +2,7 @@
 
 namespace App\Character;
 
+use App\ArmorType\ShieldType;
 use App\AttackType\CastSpellType;
 
 class MageFighter extends Character
@@ -14,5 +15,6 @@ class MageFighter extends Character
         $this->armor = 0.25;
 
         $this->attackType = new CastSpellType();
+        $this->armorType = new ShieldType();
     }
 }
