diff --git a/src/Character/Archer.php b/src/Character/Archer.php
index cdf35a3..a57ccdd 100644
--- a/src/Character/Archer.php
+++ b/src/Character/Archer.php
@@ -8,10 +8,4 @@ class Archer extends Character
     {
         parent::__construct(80, 10, 0.15);
     }
-
-    public function castSpell(): int
-    {
-        // I can't cast spells
-        return 0;
-    }
 }
diff --git a/src/Character/Character.php b/src/Character/Character.php
index 1ed06d0..25ac7fd 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -9,8 +9,6 @@ abstract class Character
     private int $currentHealth;
     private string $nickname = '';
 
-    abstract public function castSpell(): int;
-
     public function __construct(
         private int $maxHealth,
         private int $baseDamage,
diff --git a/src/Character/Fighter.php b/src/Character/Fighter.php
index c491bbe..48e31a6 100644
--- a/src/Character/Fighter.php
+++ b/src/Character/Fighter.php
@@ -8,10 +8,4 @@ class Fighter extends Character
     {
         parent::__construct(90, 12, 0.25);
     }
-
-    public function castSpell(): int
-    {
-        // I can't cast spells
-        return 0;
-    }
 }
diff --git a/src/Character/Mage.php b/src/Character/Mage.php
index c45b15d..bdd87ee 100644
--- a/src/Character/Mage.php
+++ b/src/Character/Mage.php
@@ -10,12 +10,4 @@ class Mage extends Character
     {
         parent::__construct(70, 8, 0.10);
     }
-
-    /**
-     * Damage: 3d10
-     */
-    public function castSpell(): int
-    {
-        return Dice::roll(10) + Dice::roll(10) + Dice::roll(10);
-    }
 }
