diff --git a/src/Character/Archer.php b/src/Character/Archer.php
index a57ccdd..cdf35a3 100644
--- a/src/Character/Archer.php
+++ b/src/Character/Archer.php
@@ -8,4 +8,10 @@ class Archer extends Character
     {
         parent::__construct(80, 10, 0.15);
     }
+
+    public function castSpell(): int
+    {
+        // I can't cast spells
+        return 0;
+    }
 }
diff --git a/src/Character/Fighter.php b/src/Character/Fighter.php
index 48e31a6..c491bbe 100644
--- a/src/Character/Fighter.php
+++ b/src/Character/Fighter.php
@@ -8,4 +8,10 @@ class Fighter extends Character
     {
         parent::__construct(90, 12, 0.25);
     }
+
+    public function castSpell(): int
+    {
+        // I can't cast spells
+        return 0;
+    }
 }
diff --git a/src/Character/Mage.php b/src/Character/Mage.php
index ce82c46..c45b15d 100644
--- a/src/Character/Mage.php
+++ b/src/Character/Mage.php
@@ -2,10 +2,20 @@
 
 namespace App\Character;
 
+use App\Dice;
+
 class Mage extends Character
 {
     public function __construct()
     {
         parent::__construct(70, 8, 0.10);
     }
+
+    /**
+     * Damage: 3d10
+     */
+    public function castSpell(): int
+    {
+        return Dice::roll(10) + Dice::roll(10) + Dice::roll(10);
+    }
 }
