diff --git a/src/AttackType/BowType.php b/src/AttackType/BowType.php
new file mode 100644
index 0000000..45d2ba3
--- /dev/null
+++ b/src/AttackType/BowType.php
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class BowType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        $criticalChance = Dice::roll(100);
+
+        return $criticalChance > 70 ? $baseDamage * 3 : $baseDamage;
+    }
+}
diff --git a/src/AttackType/FireBoltType.php b/src/AttackType/FireBoltType.php
new file mode 100644
index 0000000..61eac1d
--- /dev/null
+++ b/src/AttackType/FireBoltType.php
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class FireBoltType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        return $this->castFireBolt();
+    }
+
+    /**
+     * Damage: 3d10
+     */
+    private function castFireBolt(): int
+    {
+        return Dice::roll(10) + Dice::roll(10) + Dice::roll(10);
+    }
+}
diff --git a/src/AttackType/TwoHandedSwordType.php b/src/AttackType/TwoHandedSwordType.php
new file mode 100644
index 0000000..d3138a7
--- /dev/null
+++ b/src/AttackType/TwoHandedSwordType.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\AttackType;
+
+use App\Dice;
+
+class TwoHandedSwordType implements AttackType
+{
+    public function performAttack(int $baseDamage): int
+    {
+        return $baseDamage + $this->twoHandedSwordDamage();
+    }
+
+    private function twoHandedSwordDamage(): int
+    {
+        return Dice::roll(12) + Dice::roll(12);
+    }
+}
