diff --git a/src/Character/Character.php b/src/Character/Character.php
index 31791aa..21c275a 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -30,8 +30,9 @@ abstract class Character
 
     public function receiveAttack(int $damage): int
     {
-        $armorReduction = (int) ($damage * $this->armor);
-        $damageTaken = $damage - $armorReduction;
+        $armorReduction = $this->armorType->getArmorReduction($damage);
+
+        $damageTaken = max($damage - $armorReduction, 0);
         $this->currentHealth -= $damageTaken;
 
         return $damageTaken;
