diff --git a/src/Character/Character.php b/src/Character/Character.php
index 1460a34..7c2f782 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -2,8 +2,8 @@
 
 namespace App\Character;
 
+use App\ArmorType\ArmorType;
 use App\AttackType\AttackType;
-use App\Dice;
 
 abstract class Character
 {
@@ -13,8 +13,8 @@ abstract class Character
     public function __construct(
         private int $maxHealth,
         private int $baseDamage,
-        private float $armor,
-        private AttackType $attackType
+        private AttackType $attackType,
+        private ArmorType $armorType
     ) {
         $this->currentHealth = $this->maxHealth;
     }
