diff --git a/src/Character/Character.php b/src/Character/Character.php
index 6a5ab34..36aa595 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -38,8 +38,9 @@ class Character
 
     public function receiveAttack(int $damage): int
     {
-        $armorReduction = (int) ($damage * $this->armor);
-        $damageTaken = $damage - $armorReduction;
+        $armorReduction = $this->armorType->getArmorReduction($damage);
+
+        $damageTaken = max($damage - $armorReduction, 0);
         $this->currentHealth -= $damageTaken;
 
         return $damageTaken;
