diff --git a/src/Character/MageFighter.php b/src/Character/MageFighter.php
new file mode 100644
index 0000000..17202fb
--- /dev/null
+++ b/src/Character/MageFighter.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Character;
+
+use App\AttackType\FireBoltType;
+
+class MageFighter extends Character
+{
+    public function __construct()
+    {
+        parent::__construct(100, 12, 0.25, new FireBoltType());
+    }
+}
