diff --git a/src/Character/Character.php b/src/Character/Character.php
index 9a63e72..e7f47c1 100644
--- a/src/Character/Character.php
+++ b/src/Character/Character.php
@@ -2,6 +2,7 @@
 
 namespace App\Character;
 
+use App\ArmorType\ArmorType;
 use App\AttackType\AttackType;
 use App\Dice;
 
@@ -16,8 +17,8 @@ class Character
     public function __construct(
         private int $maxHealth,
         private int $baseDamage,
-        private float $armor,
-        private AttackType $attackType
+        private AttackType $attackType,
+        private ArmorType $armorType
     ) {
         $this->currentHealth = $this->maxHealth;
     }
