diff --git a/src/GameApplication.php b/src/GameApplication.php
index 3af463f..e3c2ce4 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -49,7 +49,14 @@ class GameApplication
             $aiAttackCommand->execute();
 
             if ($this->didPlayerDie($player)) {
-                return $this->endBattle($fightResultSet, $ai, $player);
+                $undoChoice = GameApplication::$printer->confirm('You died! Do you want to undo your last turn?');
+                if (!$undoChoice) {
+                    return $this->endBattle($fightResultSet, $ai, $player);
+                }
+
+                // These have to be undone in the order they were executed
+                $aiAttackCommand->undo();
+                $playerAction->undo();
             }
 
             $this->printCurrentHealth($player, $ai);
