diff --git a/src/GameApplication.php b/src/GameApplication.php
index 329859b..11acfdc 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -60,8 +60,15 @@ class GameApplication
             $aiAttackCommand->execute();
 
             if ($this->didPlayerDie($player)) {
-                $this->endBattle($fightResultSet, $ai, $player);
-                return;
+                $undoChoice = GameApplication::$printer->confirm('You died! Do you want to undo your last turn?');
+                if (!$undoChoice) {
+                    $this->endBattle($fightResultSet, $ai, $player);
+                    return;
+                }
+
+                // These have to be undone in the order they were executed
+                $aiAttackCommand->undo();
+                $playerAction->undo();
             }
 
             $this->printCurrentHealth($player, $ai);
